source/mqtt_client_connection.c

```diff
@@ -49,10 +49,6 @@ struct mqtt_connection_binding {
      * Lets us invoke callbacks on the python object without preventing the GC from cleaning it up. */
     PyObject *self_proxy;
 
-    /* To not run into a segfault calling on_close with the connection being freed before the callback
-     * can be invoked, we need to keep the PyCapsule alive. */
-    PyObject *self_capsule;
```

Comment:
`self_capsule` was a trick we used to keep the python object alive before the client tear down. As we now have the termination callback, we no longer need it.